src/VisualStudio/Core/Def/DocumentOutline/DocumentOutlineControl.xaml.cs

```diff
@@ -0,0 +1,291 @@
+            => SetSortOptionAndUpdateDataModel(SortOption.Type);
+
+        private void SetSortOptionAndUpdateDataModel(SortOption sortOption)
+        {
+            SortOption = sortOption;
+            StartFilterAndSortDataModelTask();
+        }
+
+        /// <summary>
+        /// When a symbol node in the window is clicked, move the caret to its position in the latest active text view.
+        /// </summary>
+        private void JumpToContent(object sender, EventArgs e)
+        {
+            if (sender is StackPanel panel && panel.DataContext is DocumentSymbolUIItem symbol)
+            {
+                var token = _asyncListener.BeginAsyncOperation(nameof(JumpToContentAsync));
+                var task = JumpToContentAsync(symbol);
+                task.CompletesAsyncOperation(token);
+            }
+        }
+
+        /// <summary>
+        /// Given a DocumentSymbolUIItem, moves the caret to the start of its selection range in the latest active text view.
+        /// </summary>
+        private async Task JumpToContentAsync(DocumentSymbolUIItem symbol)
+        {
+            // Switch to the UI thread to update the latest active text view.
+            await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(_cancellationToken);
+
+            var activeTextView = GetLastActiveIWpfTextView();
+            if (activeTextView is null)
+                return;
+
+            // When the user clicks on a symbol node in the window, we want to move the cursor to that line in the editor. If we
+            // don't unsubscribe from Caret_PositionChanged first, we will call StartHighlightExpandAndPresentItemsTask() once
+            // we move the cursor ourselves. This is not ideal because we would be doing extra work to update the view with an
+            // identical document symbol tree.
+            activeTextView.Caret.PositionChanged -= Caret_PositionChanged;
+
+            // Prevents us from being permanently unsubscribed if an exception is thrown while updating the text view selection.
+            try
+            {
+                // Map the symbol's selection range start SnapshotPoint to a SnapshotPoint in the current textview.
+                var currentPoint = symbol.SelectionRangeSpan.Start.TranslateTo(activeTextView.TextSnapshot, PointTrackingMode.Negative);
+
+                // Set the active text view selection to this SnapshotPoint (by converting it to a SnapshotSpan).
+                var currentSpan = new SnapshotSpan(currentPoint, currentPoint);
+                activeTextView.SetSelection(currentSpan);
+                activeTextView.ViewScroller.EnsureSpanVisible(currentSpan);
+            }
+            finally
+            {
+                // Resubscribe to Caret_PositionChanged again so that when the user clicks somewhere else, we can highlight that node.
+                activeTextView.Caret.PositionChanged += Caret_PositionChanged;
+            }
+        }
+    }
```

Comment:
this is all O(1), and we should be called on the UI thread anyways.  So i would say that this method should not be async and we should just handle jumping to content totally sychronously as this should all be cheap.

Reply:
It looks like the only asynchrony is the SwitchToMainThread, which isn't needed since the event handler is already raised there in the first place.